### PR TITLE
[e2e] Run full e2e suite against external hosts

### PR DESCRIPTION
**Pull request by Stojdza**
### Changes proposed in this Pull Request:

Part of pdV5qK-wB-p2

Following the changes in https://github.com/woocommerce/woocommerce/pull/49017 and https://github.com/woocommerce/woocommerce/pull/49403, I added several updates:
- Use global PW config and retry failed tests once in the CI - [3cbf774](https://github.com/woocommerce/woocommerce/pull/49597/commits/3cbf7742ea591ab0e9092734374705d238408d66)
- Run full suite for Pressable and WPCOM - [f20578c](https://github.com/woocommerce/woocommerce/pull/49597/commits/f20578cb46dc95843772d53e64e5c5fed5459db2)
- Trigger Pressable and WPCOM on 'release-checks' - [d864e80](https://github.com/woocommerce/woocommerce/pull/49597/commits/d864e807916f8361b244a8446295f676e753e543)

### How to test the changes in this Pull Request:

0. Find `E2E_ENV_KEY` in the secret store and export it as an environment variable. 
1. Checkout this branch 
2. Run `pnpm test:e2e:with-env default-pressable`
   - All e2e tests start running against Pressable website
3. Run `pnpm test:e2e:with-env default-wpcom`
   - All e2e tests start running against WPCOM website

### Changelog entry

-   [x] Automatically create a changelog entry from the details below.

-   [ ] This Pull Request does not require a changelog entry. (Comment required below)

<details>

<summary>Changelog Entry Details</summary>

#### Significance

-   [x] Patch
-   [ ] Minor
-   [ ] Major

#### Type

-   [ ] Fix - Fixes an existing bug
-   [ ] Add - Adds functionality
-   [x] Update - Update existing functionality
-   [ ] Dev - Development related task
-   [ ] Tweak - A minor adjustment to the codebase
-   [ ] Performance - Address performance issues
-   [ ] Enhancement - Improvement to existing functionality

#### Message 

Run full e2e tests suite against Pressable and WPCOM websites.

</details>

<details>

<summary>Changelog Entry Comment</summary>

#### Comment 

</details>

### CODE REVIEW

File: plugins/woocommerce/tests/e2e-pw/envs/default-pressable/playwright.config.js [hunk 1]
```diff
@@ -3,12 +3,11 @@ const { devices } = require( '@playwright/test' );
 
 config = {
 	...config,
-	retries: 0,
```
Comment (Stojdza) [inline plugins/woocommerce/tests/e2e-pw/envs/default-pressable/playwright.config.js:6]: Remove this, and [use global PW config](https://github.com/woocommerce/woocommerce/blob/trunk/plugins/woocommerce/tests/e2e-pw/playwright.config.js#L67) - to retry once on CI

File: plugins/woocommerce/tests/e2e-pw/envs/default-pressable/playwright.config.js [hunk 2]
```diff
@@ -3,12 +3,11 @@ const { devices } = require( '@playwright/test' );
...
 	projects: [
 		{
 			name: 'default',
 			use: { ...devices[ 'Desktop Chrome' ] },
-			testMatch: '**basic.spec.js',
+			testMatch: '**.spec.js',
```
Comment (Stojdza) [inline plugins/woocommerce/tests/e2e-pw/envs/default-pressable/playwright.config.js]: Is there some better way to run all e2e tests here?

**Review by Stojdza — COMMENTED** [review]
@adimoldovan this is a draft PR I opened for a discussion and to explore some changes to the current setup. Below are two comments/changes where I would appreciate your thoughts. Thanks! 🙇‍♂️

File: plugins/woocommerce/tests/e2e-pw/envs/default-pressable/playwright.config.js (hunk 2 above)
Comment (adimoldovan) [reply plugins/woocommerce/tests/e2e-pw/envs/default-pressable/playwright.config.js]: Removing `testMatch` entirely will run all the tests.

File: plugins/woocommerce/tests/e2e-pw/envs/default-pressable/playwright.config.js (hunk 1 above)
Comment (adimoldovan) [reply plugins/woocommerce/tests/e2e-pw/envs/default-pressable/playwright.config.js:6]: 👍🏼 Makes sense to not override the global config, I think this one was a left over from when I was trying some things out and forgot to remove it.

File: plugins/woocommerce/tests/e2e-pw/envs/default-pressable/playwright.config.js [hunk 3]
```diff
@@ -3,12 +3,13 @@ const { devices } = require( '@playwright/test' );
...
 	...config,
-	retries: 0,
 	projects: [
 		{
 			name: 'default',
 			use: { ...devices[ 'Desktop Chrome' ] },
-			testMatch: '**basic.spec.js',
+		},
+		{
+			name: 'Core API tests',
```
Comment (Stojdza) [inline plugins/woocommerce/tests/e2e-pw/envs/default-pressable/playwright.config.js]: @adimoldovan, with `name: 'Core API tests'` added here, I'm running API tests against Prressable/WPCOM. Many of them are failing, but before any further investigation, I want to check with you to see if I'm on the right path.

Screenshot 2024-07-17 at 11 32 42

Comment (adimoldovan) [reply plugins/woocommerce/tests/e2e-pw/envs/default-pressable/playwright.config.js]: 🤔 If your plan is to run API tests (`plugins/woocommerce/tests/api-core-tests`), that will not work. That will run all e2e tests (everything in `plugins/woocommerce/tests/e2e-pw/tests`).

**Review by adimoldovan — CHANGES_REQUESTED** [review]
@Stojdza As we discussed in our chat, let's only add the tests that are passing in these runs and include the rest one by one once they get fixed in follow-up PRs.

**Review by adimoldovan — APPROVED** [review]

**Pull request merged**